FreeRTOS_DNS.c

```diff
@@ -239,8 +235,38 @@
                                      struct freertos_addrinfo ** ppxAddressInfo );
     #endif
 
+/** @brief This function is called by the macro 'vSetField16'. It store
+ *         a 16-bit number in a buffer in big-endian format ( MSB first ).
+ * @param[in] pucBase: A pointer to a buffer where to store a uint16_t.
+ * @param[in] uxOffset: The offset within pucBase.
+ * @param[in] usValue: The 16-bit value to be stored.
+ */
+    void vSetField16helper( uint8_t * pucBase,
```

Comment:
I have put the Doxygen comments in the C-file in stead of the header file, because these helper functions are called indirectly through a macro, `vSetField16()` and `vSetField32()`:
~~~c
/** @brief The macros vSetField16() and vSetField32() will
 * write either a short or a 32-bit value into an array of bytes.
 * They will be stored big-endian. The helper functions do
 * the actual work..
 */
~~~
And in fact, the macros are also kind of private: they're only used by the DNS module.